packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -112,7 +112,9 @@ library LibProposing {
                 txListByteOffset: 0, // to be initialized below
                 txListByteSize: 0, // to be initialized below
                 minTier: 0, // to be initialized below
-                blobUsed: txList.length == 0
+                blobUsed: txList.length == 0,
+                parentMetaHash: state.blocks[(b.numBlocks - 1)
+                    % config.blockRingBufferSize].metaHash
```

Comment:
Having 'parentMetaHash' prevents the parent L2 block from being replaced by a new one with different txList, for eample.

Reply:
Yeah in the case of multiple L2 blocks in a single L1 block that have different proposers this would indeed be helpful. Though pretty unlikely to have multiple proposers in a single L1 block on mainnet, seems like a good thing to make available for simplicity.